### PR TITLE
Store onboarding: Update payment task description, priority and URL

### LINKED ISSUE #10570: Update payment onboarding task
- [ ] Prioritize payment task over store launch and custom domain onboarding tasks.
- [ ]  Update wording for the onboarding task to "Manage payments seamlessly with WooPayments, free from setup or monthly fees."
- [ ] Update link for onboarding task `wcPay` type - following pe5sF9-1VU-p2.

### PR DESCRIPTION
**Pull request by itsmeichigo**
Closes: #10570 

## Description

This PR updates a few things for the payment onboarding task:
- Prioritize it over the custom domain and launch store tasks.
- New description following 7abnJ5szUDhwiczNwhcyW3-fi
- Updated the URL to open for `woocommerce-payment` task following pe5sF9-1VU-p2.

## Testing instructions

- Log in to the app.
- Open a free trial site or create one if needed.
- Navigate to the My Store screen, notice the payment task on the onboarding task list is above the custom domain and launch store tasks. The description should be correct.
- Tap the task > Continue Setup. A web view should be displayed showing the WooPayments page with the Finish Setup button displayed.

## Screenshots

 

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Networking/Networking/Model/StoreOnboardingTask.swift [hunk 1]
```diff
@@ -60,11 +60,11 @@ private extension StoreOnboardingTask.TaskType {
             return 1
         case .addFirstProduct:
             return 2
-        case .launchStore:
+        case .payments, .woocommercePayments:
```
Comment (selanthiraiyan) [inline Networking/Networking/Model/StoreOnboardingTask.swift:63]: Nit: Kindly consider adding a unit test to check the sort order.

**Review by selanthiraiyan — APPROVED** [review]
This works as expected. 🚀

File: Networking/Networking/Model/StoreOnboardingTask.swift (hunk 1 above)
Comment (itsmeichigo) [reply Networking/Networking/Model/StoreOnboardingTask.swift:63]: These are already tested in `StoreOnboardingViewModelTests` and failed - updated in 2d56e7b2e506f4928895c5f63cc684b7128b4197.

**Comment by itsmeichigo** [comment]
I added a few more changes:
- Changed the URL for the `woocommerce-payments` task following pe5sF9-1Xb-p2#comment-2813.
- Restored the description for the `payment` onboarding task.

Please feel free to test again @selanthiraiyan, I'll merge this PR tomorrow if there's no concern about these.

**Comment by selanthiraiyan** [comment]
The new commits look good to me. 🚀

**Pull request merged**